### ISSUE TITLE
No runs in parallel

### ISSUE THREAD

**Issue by ideivitcom** [Netflix/metaflow#165]
Hi all.
I need to process in parallel a huge quantity of data. I'm testing with metaflow and i have the results in the attached file (i attach the code too)
As we can observe, the counter is perfect, normally in a parallel execution the counter is mixed between the proceses or threads.
As we can observe too, we can see as when the task finished, start the other.
What im doing wrong?
[output_metaflow.txt](https://github.com/Netflix/metaflow/files/4388574/output_metaflow.txt)

[PruebaParaleloFlow.py.txt](https://github.com/Netflix/metaflow/files/4388578/PruebaParaleloFlow.py.txt)

**Comment by ideivitcom** [comment]
with a random time:
[output_metaflow_random.txt](https://github.com/Netflix/metaflow/files/4388922/output_metaflow_random.txt)

 print("processing {}".format(i))
            time.sleep(random.randint(1, 3))            
            i+=1

**Comment by tuulos** [comment]
@ideivitcom your flow works correctly (the tasks are run concurrently) but there's some buffering for logging that swallows your small messages during execution and also shows them in the end.

You can verify the correct behavior in two ways:
 1) `print(time.time())` in the task - you will see that the timestamps are overlapping
 2) make the message larger (e.g. `print('%s count %d' % ('-' * 1000, i))`) which forces the buffer to be flushed earlier.

The buffering behavior is confusing in your case. I will open a separate ticket to address the issue in the long term.

**Comment by ideivitcom** [comment]
Thank you!!
